lfsapi/client.go

```diff
@@ -85,6 +85,13 @@ func (c *Client) Do(req *http.Request) (*http.Response, error) {
 	req.Header = c.extraHeadersFor(req)
 	req.Header.Set("User-Agent", UserAgent)
 
+	return c.do(req)
+}
+
+// do performs an *http.Request respecting redirects, and handles the response
+// as defined in c.handleResponse. Notably, it does not alter the headers for
+// the request argument in any way.
+func (c *Client) do(req *http.Request) (*http.Response, error) {
```

Comment:
I think the `User-Agent` bit should move to `do()` too. This ensures a consistent user agent for all outgoing http requests.

Reply:
Nice -- I picked this up in https://github.com/git-lfs/git-lfs/pull/2733/commits/bda2fe845fd7b1b4917ac266c17cea8ec055261a.